src/validation.cpp

```diff
@@ -1593,6 +1593,13 @@ static unsigned int GetBlockScriptFlags(const CBlockIndex& block_index, const Co
     {
         // Enforce WITNESS rules whenever P2SH is in effect
         flags |= SCRIPT_VERIFY_P2SH | SCRIPT_VERIFY_WITNESS;
+        // Enforce Taproot (BIP340-BIP342) whenever WITNESS is in effect,
+        // unless except
+        if (!consensusparams.TaprootException.IsNull() ||                        // no Taproot exception on this chain
```

Comment:
Did you mean to drop the `!`?

Reply:
Ah yes, good catch!

Maybe my preference would be to drop the `IsNull` check and purely rely on the the actual comparison, which is already sufficient and necessary.

Reply:
It's nice for it to be identifcal to the BIP16 check, which made this mistake easier to see. It's also probably more similar to what future code will look like that has to handle multiple blocks.